public/app/features/query/state/PanelQueryRunner.ts

```diff
@@ -370,13 +370,14 @@ async function getDataSource(
   scopedVars: ScopedVars,
   publicDashboardAccessToken?: string
 ): Promise<DataSourceApi> {
-  if (publicDashboardAccessToken) {
-    return new PublicDashboardDataSource(datasource);
+  if (!publicDashboardAccessToken && datasource && (datasource as DataSourceApi).query) {
+    return datasource as DataSourceApi;
   }
 
-  if (datasource && (datasource as any).query) {
-    return datasource as DataSourceApi;
+  const ds = await getDatasourceSrv().get(datasource, scopedVars);
```

Comment:
I could have sworn you can't use getDatasourceSrv() while unauthed. But I tested it out and it works fine. I was smokin beers I guess 🤷‍♂️

Reply:
You weren't smoking beers! Tested it on an anonymous window and the page doesn't load.. so turns out we are not allowed to call this... I'm wondering how can we get the interval then.

Reply:
Weird, I just did the same thing and it works fine. I'll rebuild stuff and see if that changes it

Reply:
This just got a lot more complicated.

Reply:
Fixed on https://github.com/grafana/grafana/pull/55489/commits/f31996dfab174d8a008e9f22c10c671ec961b4e1